apps/rewards/src/invite/inviteReward.service.ts

```diff
@@ -0,0 +1,171 @@
+          this.inviterHasNotReachedWeeklyLimit(inviter),
+          this.inviteeRewardNotInProgress(invitee)
+        ])
+        if (conditions.every(condition => condition)) {
+          this.logger.log(
+            `Starting to send reward to ${inviter} for inviting ${invitee}`
+          )
+          try {
+            const inviteReward = await this.createInviteReward(inviter, invitee)
+            this.sendInviteReward(inviteReward)
+          } catch (error) {
+            this.logger.log(`Error creating and sending reward ${error}`)
+          }
+        }
+      }
+      return maxBlock
+    })
+
+    this.isRunning = false
+  }
+
+  async usingLastNotifiedBlock(fn: (blockNumber: number) => Promise<number>) {
+    try {
+      let lastNotifiedBlock = await this.notifiedBlockRepository.findOne({
+        key: 'inviteReward'
+      })
+      if (!lastNotifiedBlock) {
+        lastNotifiedBlock = {
+          id: uuidv4(),
+          key: 'inviteReward',
+          // TODO: Replace this by the correct starting block in alfajores and mainnet.
+          // For now, putting a high value to avoid potential issues if this is accidentally deployed.
+          blockNumber: 100000000
+        }
+      }
+      const fromBlock = lastNotifiedBlock.blockNumber + 1
+      const lastBlock = await fn(fromBlock)
+      if (lastBlock > fromBlock) {
+        await this.notifiedBlockRepository.save(
+          NotifiedBlock.of({
+            id: lastNotifiedBlock.id,
+            key: lastNotifiedBlock.key,
+            blockNumber: lastBlock
+          })
+        )
+      }
+    } catch (error) {
+      this.logger.log(`Error while calculating invite rewards: ${error}`)
+    }
+  }
+
+  isKomenciSender(address: string) {
+    return this.komenciAddresses.includes(address.toLowerCase())
+  }
+
+  async addressIsConsideredVerified(address: string, identifier: string) {
+    const attestations = await this.contractKit.contracts.getAttestations()
+    const attestationStat = await attestations.getAttestationStat(
+      identifier,
+      address
+    )
+    return isAccountConsideredVerified(attestationStat).isVerified
+  }
+
+  async inviterHasNotReachedWeeklyLimit(inviter: string) {
+    const [_, invites] = await this.inviteRewardRepository.findAndCount({
+      where: {
+        inviter,
+        createdAt: Raw(alias => `${alias} >= date_trunc('week', current_date)`)
+      }
+    })
+    return invites < WEEKLY_INVITE_LIMIT
+  }
+
+  async inviteeRewardNotInProgress(invitee: string) {
+    const reward = await this.inviteRewardRepository.findOne({ invitee })
+    return !reward
+  }
+
+  async createInviteReward(inviter: string, invitee: string) {
+    const inviteReward = InviteReward.of({
+      id: uuidv4(),
+      inviter,
+      invitee,
+      state: RewardStatus.Created,
+      createdAt: new Date(Date.now()).toISOString()
+    })
+    return this.inviteRewardRepository.save(inviteReward)
```

Comment:
This will fail if the invitee already exists in any record of the DB, correct? Can you note this behavior in a header comment?